File: hw/aimacana/u3/hw25CompositePattern/javascript/model/Supervisor.js
const Employee = require('./Employee');

class Supervisor extends Employee {
    constructor() {
        super();
        this.directReports = [];
    }
    
    stateName() {
        let result = super.stateName();
        if (this.directReports.length > 0) {
            for (let i = 0; i < this.directReports.length; ++i) {
                result += this.directReports[i].stateName();
            }
        }
        return result;
    }
    
    add(anEmployee) {
        this.directReports.push(anEmployee);
    }
}
module.exports = Supervisor;

File: hw/aimacana/u3/hw25CompositePattern/python/model/supervisor.py
from .employee import Employee

class Supervisor(Employee):
    def __init__(self):
        super().__init__()
        self.direct_reports = []
        
    def state_name(self):
        result = super().state_name()
        if len(self.direct_reports) > 0:
            for report in self.direct_reports:
                result += report.state_name()
        return result
                
    def add(self, an_employee):
        self.direct_reports.append(an_employee)
